src/components/Separator/Separator.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import styled from "styled-components";
+import * as RadixSeparator from "@radix-ui/react-separator";
+interface Props extends RadixSeparator.SeparatorProps {
+  size: "xs" | "sm" | "md" | "lg" | "xl" | "xxl";
+}
+const CUISeparator = styled(RadixSeparator.Root)<Props>`
+  &[data-orientation="horizontal"] {
+    height: 0.0625rem;
+    width: 100%;
+  }
+  &[data-orientation="vertical"] {
+    height: 100%;
+    width: 0.0625rem;
+  }
+  ${({ theme, size, orientation }) => `
+    background: ${theme.click.separator.color.stroke.default};
+    margin: ${
+      orientation === "horizontal"
+        ? theme.click.separator.horizontal.space.y[size]
+        : theme.click.separator.vertical.space.y.all
+    } ${
+    orientation === "horizontal"
+      ? theme.click.separator.horizontal.space.x.all
+      : theme.click.separator.vertical.space.x[size]
```

Comment:
i think this would be more readable as

```js
${({ theme, size, orientation }) => `
    background: ${theme.click.separator.color.stroke.default};
    margin: ${
      orientation === "horizontal"
        ? `${theme.click.separator.horizontal.space.y[size]} ${theme.click.separator.horizontal.space.x.all}`
        : `${theme.click.separator.vertical.space.y.all} ${theme.click.separator.vertical.space.x[size]}`
    } 
  `}
```